sensors/triggers/kafka/kafka.go

```diff
@@ -194,14 +199,20 @@ func (t *KafkaTrigger) Execute(ctx context.Context, events map[string]*v1alpha1.
 		pk = trigger.URL
 	}
 
-	t.Producer.Input() <- &sarama.ProducerMessage{
+	msg := &sarama.ProducerMessage{
 		Topic:     trigger.Topic,
-		Key:       sarama.StringEncoder(pk),
 		Value:     sarama.ByteEncoder(payload),
-		Partition: trigger.Partition,
 		Timestamp: time.Now().UTC(),
 	}
 
+	if trigger.Partition == -1 && trigger.PartitioningKey == "" {
```

Comment:
In terms of readability, this condition could be moved closer to definition of `pk` since `trigger.PartitioningKey == ""` is already being checked.